### PR TITLE
[API] Password reset

### PR DESCRIPTION
**Pull request by arti0090**
| Q               | A
| --------------- | -----
| Branch?         | master 
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Api/EmailContext.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    /** @var EmailCheckerInterface */
+    private $emailChecker;
+
+    /** @var TranslatorInterface */
+    private $translator;
+
+    public function __construct(
+        EmailCheckerInterface $emailChecker,
+        TranslatorInterface $translator
+    ) {
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/EmailContext.php:23]: Could be in one line I guess

File: src/Sylius/Bundle/ApiBundle/Command/ResetPasswordRequest.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Command;
+
+class ResetPasswordRequest implements ChannelCodeAwareInterface, LocaleCodeAwareInterface
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Command/ResetPasswordRequest.php:7]:
```suggestion
class RequestPasswordReset implements ChannelCodeAwareInterface, LocaleCodeAwareInterface
```
or simply
```suggestion
class ResetPassword implements ChannelCodeAwareInterface, LocaleCodeAwareInterface
```

File: src/Sylius/Bundle/ApiBundle/CommandHandler/ResetPasswordRequestHandler.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        $user->setPasswordResetToken($this->generator->generate());
+        $user->setPasswordRequestedAt(new \DateTime());
+
+        $this->eventBus->dispatch(
+            new ResetPasswordRequested($command->getEmail(), $command->getChannelCode(), $command->getLocaleCode()),
+            [new DispatchAfterCurrentBusStamp()
+        ]);
+
+
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/CommandHandler/ResetPasswordRequestHandler.php:49]:
```suggestion
```

File: src/Sylius/Bundle/ApiBundle/Event/ResetPasswordRequested.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+class ResetPasswordRequested
+{
+    /** @var string */
+    public $email;
+
+    /** @var string */
+    private $channelCode;
+
+    /** @var string */
+    private $localeCode;
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Event/ResetPasswordRequested.php:16]: Could be protected

File: src/Sylius/Bundle/ApiBundle/EventHandler/ResetPasswordRequestedHandler.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    public function __construct(MessageBusInterface $commandBus)
+    {
+        $this->commandBus = $commandBus;
+    }
+
+    public function __invoke(ResetPasswordRequested $resetPasswordRequested): void
+    {
+        $this->commandBus->dispatch(
+            new SendResetPasswordEmail(
+                $resetPasswordRequested->email,
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/EventHandler/ResetPasswordRequestedHandler.php:25]:
```suggestion
                $resetPasswordRequested->email(),
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/ShopUser.xml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+<resources xmlns="https://api-platform.com/schema/metadata"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="https://api-platform.com/schema/metadata https://api-platform.com/schema/metadata/metadata-2.0.xsd"
+>
+    <resource class="%sylius.model.shop_user.class%" shortName="ShopUser">
+        <attribute name="validation_groups">sylius</attribute>
+
+        <collectionOperations>
+            <collectionOperation name="shop_password_reset_request">
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/ShopUser.xml:22]: I would be for moving this endpoint to Customer resource

**Comment by arti0090** [comment]
TODO for part 2:

- [ ] Add phpspecs to new classes
- [ ] if possible update swagger docs with properties that are needed for password reset
- [ ] add endpoint for change the password with token from email
- [ ] implement rest of behat scenarios

File: src/Sylius/Behat/Client/ApiSecurityClientInterface.php [hunk 7]
```diff
@@ -13,6 +13,8 @@
 
 namespace Sylius\Behat\Client;
 
+use Symfony\Component\HttpFoundation\Response;
```
Comment (arti0090) [inline src/Sylius/Behat/Client/ApiSecurityClientInterface.php:16]: will remove with next commit

File: src/Sylius/Behat/Context/Api/Shop/LoginContext.php [hunk 8]
```diff
@@ -111,4 +165,29 @@ public function iShouldBeNotifiedAboutBadCredentials(): void
...
+    /**
+     * @Then I should be notified that the :elementName is required
+     */
+    public function iShouldBeNotifiedThatFirstNameIsRequired(string $elementName): void
+    {
+        Assert::contains(
+            $this->responseChecker->getError($this->apiClient->getLastResponse()),
+            sprintf('Please enter your %s.', $elementName)
+        );
+    }
```
Comment (arti0090) [inline src/Sylius/Behat/Context/Api/Shop/LoginContext.php:187]: i will remove it, its not needed

File: src/Sylius/Behat/Context/Api/EmailContext.php [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+declare(strict_types=1);
+
+namespace Sylius\Behat\Context\Api;
+
+use Behat\Behat\Context\Context;
+use Sylius\Component\Core\Test\Services\EmailCheckerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Webmozart\Assert\Assert;
+
+class EmailContext implements Context
```
Comment (Tomanhez) [inline src/Sylius/Behat/Context/Api/EmailContext.php:12]:
```suggestion
final class EmailContext implements Context
```

File: src/Sylius/Bundle/ApiBundle/Command/ResetPassword.php [hunk 10]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Command;
+
+class ResetPassword implements ChannelCodeAwareInterface, LocaleCodeAwareInterface
```
Comment (Tomanhez) [inline src/Sylius/Bundle/ApiBundle/Command/ResetPassword.php:7]:
```suggestion
final class ResetPassword implements ChannelCodeAwareInterface, LocaleCodeAwareInterface
```

File: src/Sylius/Bundle/ApiBundle/Command/SendResetPasswordEmail.php [hunk 11]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Command;
+
+class SendResetPasswordEmail
```
Comment (Tomanhez) [inline src/Sylius/Bundle/ApiBundle/Command/SendResetPasswordEmail.php:7]:
```suggestion
final class SendResetPasswordEmail
```

File: src/Sylius/Bundle/ApiBundle/CommandHandler/ResetPasswordHandler.php [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+use Sylius\Bundle\ApiBundle\Command\ResetPassword;
+use Sylius\Bundle\ApiBundle\Event\ResetPasswordRequested;
+use Sylius\Component\User\Repository\UserRepositoryInterface;
+use Sylius\Component\User\Security\Generator\GeneratorInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Messenger\Stamp\DispatchAfterCurrentBusStamp;
+use Webmozart\Assert\Assert;
+
+class ResetPasswordHandler
```
Comment (Tomanhez) [inline src/Sylius/Bundle/ApiBundle/CommandHandler/ResetPasswordHandler.php:15]:
```suggestion
final class ResetPasswordHandler
```

File: src/Sylius/Bundle/ApiBundle/CommandHandler/ResetPasswordHandler.php [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+        $user = $this->userRepository->findOneByEmail($command->getEmail());
+        Assert::notNull($user);
+
+        $user->setPasswordResetToken($this->generator->generate());
+        $user->setPasswordRequestedAt(new \DateTime());
+
+        $this->eventBus->dispatch(
+            new ResetPasswordRequested($command->getEmail(), $command->getChannelCode(), $command->getLocaleCode()),
+            [new DispatchAfterCurrentBusStamp()
+        ]);
```
Comment (Tomanhez) [inline src/Sylius/Bundle/ApiBundle/CommandHandler/ResetPasswordHandler.php:47]:
```suggestion
            [new DispatchAfterCurrentBusStamp()]
       );
```

File: src/Sylius/Bundle/ApiBundle/CommandHandler/SendResetPasswordEmailHandler.php [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+namespace Sylius\Bundle\ApiBundle\CommandHandler;
+
+use Sylius\Bundle\ApiBundle\Command\SendResetPasswordEmail;
+use Sylius\Bundle\CoreBundle\Mailer\Emails;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Sylius\Component\Mailer\Sender\SenderInterface;
+use Sylius\Component\User\Repository\UserRepositoryInterface;
+
+class SendResetPasswordEmailHandler
```
Comment (Tomanhez) [inline src/Sylius/Bundle/ApiBundle/CommandHandler/SendResetPasswordEmailHandler.php:13]:
```suggestion
final class SendResetPasswordEmailHandler
```

File: src/Sylius/Bundle/ApiBundle/CommandHandler/SendResetPasswordEmailHandler.php [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+        $user = $this->userRepository->findOneByEmail($command->email);
+        $channel = $this->channelRepository->findOneByCode($command->channelCode());
+
+        $this->emailSender->send(
+            Emails::PASSWORD_RESET,
+            [$command->email],
+            [
+                'user' => $user,
+                'localeCode' => $command->localeCode(),
+                'channel' => $channel
```
Comment (Tomanhez) [inline src/Sylius/Bundle/ApiBundle/CommandHandler/SendResetPasswordEmailHandler.php:45]:
```suggestion
                'channel' => $channel,
```

File: src/Sylius/Bundle/ApiBundle/Event/ResetPasswordRequested.php [hunk 16]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Event;
+
+class ResetPasswordRequested
```
Comment (Tomanhez) [inline src/Sylius/Bundle/ApiBundle/Event/ResetPasswordRequested.php:7]:
```suggestion
final class ResetPasswordRequested
```

File: src/Sylius/Bundle/ApiBundle/EventHandler/ResetPasswordRequestedHandler.php [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\EventHandler;
+
+use Sylius\Bundle\ApiBundle\Command\SendResetPasswordEmail;
+use Sylius\Bundle\ApiBundle\Event\ResetPasswordRequested;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+class ResetPasswordRequestedHandler
```
Comment (Tomanhez) [inline src/Sylius/Bundle/ApiBundle/EventHandler/ResetPasswordRequestedHandler.php:11]:
```suggestion
final class ResetPasswordRequestedHandler
```

File: src/Sylius/Behat/Context/Api/EmailContext.php (hunk 9 above)
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/EmailContext.php:12]: Don't we have any email context classes already?

File: src/Sylius/Bundle/ApiBundle/Command/ResetPassword.php (hunk 10 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/ApiBundle/Command/ResetPassword.php:7]: Commands should be left without final, as the cost of maintenance is pretty low and it allows for easier extensibility for third-party users

Comment (lchrusciel) [reply src/Sylius/Bundle/ApiBundle/Command/ResetPassword.php:7]:
```suggestion
class RequestResetPasswordToken implements ChannelCodeAwareInterface, LocaleCodeAwareInterface
```

File: src/Sylius/Bundle/ApiBundle/Event/ResetPasswordRequested.php (hunk 16 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/ApiBundle/Event/ResetPasswordRequested.php:7]: Events can be left without final as well

File: src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Commands/Account/ResetPassword.xml [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+
+-->
+
+<serializer xmlns="http://symfony.com/schema/dic/serializer-mapping"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/serializer-mapping https://symfony.com/schema/dic/serializer-mapping/serializer-mapping-1.0.xsd"
+>
+    <class name="Sylius\Bundle\ApiBundle\Command\ResetPassword">
+        <attribute name="email">
+            <group>shop:customer:password_reset</group>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Commands/Account/ResetPassword.xml:20]:
```suggestion
            <group>shop:password_reset:create</group>
```

File: src/Sylius/Behat/Resources/config/services/contexts/api.xml [hunk 19]
```diff
@@ -18,5 +18,6 @@
     <imports>
         <import resource="api/admin.xml" />
         <import resource="api/shop.xml" />
+        <import resource="api/email.xml" />
```
Comment (GSadee) [inline src/Sylius/Behat/Resources/config/services/contexts/api.xml:21]:
```suggestion
        <import resource="api/**/*.xml" />
    </imports>
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Customer.xml [hunk 20]
```diff
@@ -33,6 +33,25 @@
                 <attribute name="input">Sylius\Bundle\ApiBundle\Command\RegisterShopUser</attribute>
                 <attribute name="output">false</attribute>
             </collectionOperation>
+
+            <collectionOperation name="shop_password_reset_request">
+                <attribute name="method">POST</attribute>
+                <attribute name="path">shop/password-reset-request</attribute>
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Customer.xml:39]:
```suggestion
                <attribute name="path">shop/request-reset-password</attribute>
```

File: src/Sylius/Behat/Context/Api/EmailContext.php (hunk 9 above)
Comment (arti0090) [reply src/Sylius/Behat/Context/Api/EmailContext.php:12]: We have but they are UI contexts :/

Comment (arti0090) [reply src/Sylius/Behat/Context/Api/EmailContext.php:21]: I will leave this in this PR as it is and will make a separate PR with changes. I will extract EmailContext and make this changes so the sylius code would be a better place for everyone 🌈

Comment (arti0090) [reply src/Sylius/Behat/Context/Api/EmailContext.php:21]: I will fix/change it with refactor of EmailContext ( see comment https://github.com/Sylius/Sylius/pull/12360#discussion_r579044089 )

File: src/Sylius/Bundle/ApiBundle/Command/SendResetPasswordEmail.php [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+{
+    /** @var string */
+    public $email;
+
+    /** @var string */
+    private $channelCode;
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Command/SendResetPasswordEmail.php:22]:
```suggestion
    public $channelCode;
```

File: src/Sylius/Bundle/ApiBundle/Command/SendResetPasswordEmail.php [hunk 22]
```diff
@@ -0,0 +1,51 @@
...
+
+    /** @var string */
+    private $localeCode;
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Command/SendResetPasswordEmail.php:25]:
```suggestion
    public $localeCode;
```

File: src/Sylius/Bundle/ApiBundle/Event/ResetPasswordRequested.php [hunk 23]
```diff
@@ -0,0 +1,51 @@
...
+{
+    /** @var string */
+    public $email;
+
+    /** @var string */
+    private $channelCode;
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Event/ResetPasswordRequested.php:22]:
```suggestion
    public $channelCode;
```

File: src/Sylius/Bundle/ApiBundle/Event/ResetPasswordRequested.php [hunk 24]
```diff
@@ -0,0 +1,51 @@
...
+
+    /** @var string */
+    protected $localeCode;
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Event/ResetPasswordRequested.php:25]:
```suggestion
    public $localeCode;
```

File: src/Sylius/Bundle/CoreBundle/Resources/views/Email/passwordReset.html.twig [hunk 25]
```diff
@@ -0,0 +1,22 @@
+{% extends '@SyliusCore/Email/layout.html.twig' %}
+
+{% block subject %}
+    {{ 'sylius.email.password_reset.subject'|trans({}, null, localeCode) }}
+{% endblock %}
+
+{% block content %}
+    {% set url = channel.hostname is not null ? 'http://' ~ channel.hostname ~ path('sylius_shop_password_reset', { 'token': user.passwordResetToken}) : url('sylius_shop_password_reset', { 'token': user.passwordResetToken}) %}
```
Comment (AdamKasp) [inline src/Sylius/Bundle/CoreBundle/Resources/views/Email/passwordReset.html.twig:8]: we can't couple `coreBundle` and `ShopBundle`

**Review by AdamKasp — CHANGES_REQUESTED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Commands/Account/ResetPassword.xml [hunk 26]
```diff
@@ -0,0 +1,26 @@
...
+
+-->
+
+<serializer xmlns="http://symfony.com/schema/dic/serializer-mapping"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/serializer-mapping https://symfony.com/schema/dic/serializer-mapping/serializer-mapping-1.0.xsd"
+>
+    <class name="Sylius\Bundle\ApiBundle\Command\ResetPassword">
+        <attribute name="email">
+            <group>shop:password_reset:create</group>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Commands/Account/ResetPassword.xml:20]:
```suggestion
            <group>shop:reset_password:create</group>
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/services/command_handlers.xml [hunk 27]
```diff
@@ -102,6 +102,13 @@
             <tag name="messenger.message_handler" bus="sylius_default.bus"/>
         </service>
 
+        <service id="Sylius\Bundle\ApiBundle\CommandHandler\RequestResetPasswordTokenHandler">
+            <argument type="service" id="sylius.repository.shop_user" />
+            <argument type="service" id="messenger.default_bus" />
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/services/command_handlers.xml:107]: Should we inject messenger bus?

File: src/Sylius/Bundle/ApiBundle/Command/LocaleCodeAwareInterface.php [hunk 28]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Command/LocaleCodeAwareInterface.php:3]: missing license block

File: src/Sylius/Bundle/ApiBundle/Command/LocaleCodeAwareInterface.php [hunk 29]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace Sylius\Bundle\ApiBundle\Command;
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Command/LocaleCodeAwareInterface.php:5]:
```suggestion
/** @experimental */
namespace Sylius\Bundle\ApiBundle\Command;
```

Comment (AdamKasp) [reply src/Sylius/Bundle/ApiBundle/Command/LocaleCodeAwareInterface.php:5]: and in the following files

File: src/Sylius/Bundle/ApiBundle/Resources/config/validation/ResetPassword.xml [hunk 30]
```diff
@@ -0,0 +1,29 @@
...
+    <class name="Sylius\Bundle\ApiBundle\Command\RequestResetPasswordToken">
+        <property name="email">
+            <constraint name="NotBlank">
+                <option name="message">sylius.customer.email.not_blank</option>
+                <option name="groups">
+                    <value>sylius</value>
+                </option>
+            </constraint>
+        </property>
+
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Resources/config/validation/ResetPassword.xml:27]:
```suggestion
```

File: src/Sylius/Bundle/ApiBundle/Tests/CommandHandler/SendResetPasswordEmailHandlerTest.php [hunk 31]
```diff
@@ -0,0 +1,141 @@
...
+use Sylius\Component\Core\Test\Services\EmailChecker;
+use Sylius\Component\User\Model\UserInterface;
+use Sylius\Component\User\Repository\UserRepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final class SendResetPasswordEmailHandlerTest extends KernelTestCase
+{
+
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Tests/CommandHandler/SendResetPasswordEmailHandlerTest.php:30]:
```suggestion
```

File: src/Sylius/Bundle/ApiBundle/Tests/CommandHandler/SendResetPasswordEmailHandlerTest.php [hunk 32]
```diff
@@ -0,0 +1,141 @@
...
+            $channelRepository->reveal(),
+            $userRepository->reveal()
+        );
+
+        $resetPasswordEmailHandler(new SendResetPasswordEmail(
+            'user@example.com',
+            'CHANNEL_CODE',
+            'en_US'
+        ));
+
```
Comment (AdamKasp) [inline src/Sylius/Bundle/ApiBundle/Tests/CommandHandler/SendResetPasswordEmailHandlerTest.php:77]:
```suggestion
```

**Review by AdamKasp — APPROVED** [review]

File: src/Sylius/Behat/Context/Api/Shop/LoginContext.php [hunk 33]
```diff
@@ -14,17 +14,34 @@
...
+
+    /** @var Request|null */
+    private $request;
+
+    public function __construct(
+        ApiSecurityClientInterface $client,
+        ApiClientInterface $apiClient,
+        ResponseCheckerInterface $responseChecker
+    ) {
         $this->client = $client;
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Api/Shop/LoginContext.php:42]:
```suggestion
        $this->apiSecurityClient = $apiSecurityClient;
```
?

File: src/Sylius/Bundle/ApiBundle/Tests/CommandHandler/SendResetPasswordEmailHandlerTest.php [hunk 34]
```diff
@@ -0,0 +1,139 @@
...
+        self::assertSame(1, $emailChecker->countMessagesTo('user@example.com'));
+
+        self::assertTrue($emailChecker->hasMessageTo(
+                $translator->trans('sylius.email.password_reset.to_reset_your_password_token', [], null, 'en_US'),
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Tests/CommandHandler/SendResetPasswordEmailHandlerTest.php:80]: Wrong indentation

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/PasswordReset.xml [hunk 35]
```diff
@@ -0,0 +1,46 @@
...
+>
+    <resource class="Sylius\Bundle\ApiBundle\Command\RequestResetPasswordToken" shortName="PasswordReset">
+
+    <attribute name="validation_groups">sylius</attribute>
+
+        <collectionOperations>
+            <collectionOperation name="shop_password_reset_request">
+                <attribute name="method">POST</attribute>
+                <attribute name="path">shop/request-reset-password</attribute>
+                <attribute name="messenger">input</attribute>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/PasswordReset.xml:26]: Could go to the generic section (below `validation_groups`)

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/PasswordReset.xml [hunk 36]
```diff
@@ -0,0 +1,46 @@
...
+                <attribute name="input">Sylius\Bundle\ApiBundle\Command\RequestResetPasswordToken</attribute>
+                <attribute name="output">false</attribute>
+                <attribute name="status">202</attribute>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/PasswordReset.xml:29]: Maybe could go to the generic section (below `validation_groups`)

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/PasswordReset.xml [hunk 37]
```diff
@@ -0,0 +1,46 @@
...
+                <attribute name="validation_groups">
+                    <attribute>sylius</attribute>
+                    <attribute>sylius_shop_password_reset</attribute>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/PasswordReset.xml:32]: is this group needed?

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, @arti0090! :1st_place_medal:

**Pull request merged**